ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +479,22 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestIPAServerInstallUnattendedMode(IntegrationTest):
+    """
+    Test ipa-server-install when mandatory params not specified.
+
+    When installing ipa-server in unattended mode (i.e -U option),
+    some manadatory params should be specified with the install
+    commands like -p, -r and -a etc. If we don't specify these
+    params, installation will fail.
+    """
+    def test_install_ipa_server_unattended_without_mandatory_param(self):
+
+        args = ['ipa-server-install', '-U']
+        result = self.master.run_command(args, raiseonerr=False)
```

Comment:
Multiple combinations could be tested:
- when none of -p -r -a is provided (what the test is currently doing)
- when one of -p -r -a is missing (for instance -p Secret123 -a Secret123)
- when only one of -p -r or -a is provided (for instance -p Secret123)
You can use itertools.combinations to create all the possible combinations and pytest.mark.parametrize with something like
```
@pytest.mark.parametrize(
    "additional_args",
    [ opts for l in range(0,3) for opts in itertools.combinations(mandatory_options, l)])
def test_install_ipa_server_unattended_without_mandatory_param(self, additional_args):
```
then add all the ements from additional_args to args (don't forget to add a value for each option) before calling run_command.